Dockerfile

```
@@ -21,13 +15,35 @@ COPY pyproject.toml .
 #    since setuptools-scm cannot infer the version
 RUN echo '[tool.setuptools.package-data]\n"*" = ["*"]' >> pyproject.toml
 ARG SETUPTOOLS_SCM_PRETEND_VERSION_FOR_RAGNA
-RUN pip install --progress-bar=off --no-deps .
+ARG ENVIRONMENT="default-all-py312"
+
+RUN <<EOF
+pixi shell-hook --frozen -e $ENVIRONMENT -s bash > /shell-hook.sh;
+echo "#!/bin/bash" > /entrypoint.sh;
+cat /shell-hook.sh >> /entrypoint.sh;
+echo 'exec "$@"' >> /entrypoint.sh;
+EOF
+
+RUN pixi install --frozen -e $ENVIRONMENT
+
+# Pre-download the default embedding model
+RUN pixi run -e $ENVIRONMENT python -c "from chromadb.utils.embedding_functions import ONNXMiniLM_L6_V2; ONNXMiniLM_L6_V2()._download_model_if_not_exists()"
+
+FROM debian:bookworm-slim AS production
+
+RUN useradd --create-home --shell "$(which bash)" ragna
+USER ragna
```

Our default user is 
```suggestion
USER bodil
```

See

https://github.com/Quansight/ragna/blob/9cdf0f83a729010ab690ff97daf4ba84fd389caa/ragna/_utils.py#L164-L169